exporter/kafkaexporter/kafka_exporter.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2020 The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kafkaexporter
+
+import (
+	"context"
+
+	"github.com/Shopify/sarama"
+	"go.uber.org/zap"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer/consumererror"
+	"go.opentelemetry.io/collector/consumer/pdata"
+)
+
+// kafkaProducer uses sarama to produce messages to Kafka.
+type kafkaProducer struct {
+	producer   sarama.SyncProducer
+	topic      string
+	marshaller marshaller
+	logger     *zap.Logger
+}
+
+// newExporter creates Kafka exporter.
+func newExporter(config Config, params component.ExporterCreateParams) (*kafkaProducer, error) {
+	c := sarama.NewConfig()
+	// These setting are required by the sarama implementation.
+	c.Producer.Return.Successes = true
+	c.Producer.Return.Errors = true
+	// Waits for only the local commit to succeed before responding.
+	c.Producer.RequiredAcks = sarama.WaitForLocal
+	c.Metadata.Full = config.Metadata.Full
+	c.Metadata.Retry.Max = config.Metadata.Retry.Max
+	c.Metadata.Retry.Backoff = config.Metadata.Retry.BackOff
```

Comment:
For consistency with other protocols, I would suggest to use this https://github.com/open-telemetry/opentelemetry-collector/blob/master/exporter/exporterhelper/queued_retry.go logic to control queuing and retry and disable native retries from the sarama client.

Reply:
See my comment above https://github.com/open-telemetry/opentelemetry-collector/pull/1439#discussion_r462073291